File: PHP/src/Entity/Transaction.php
<?php

namespace App\Entity;

class Transaction
{
    private ?int $idTransaction = null;
    private float $montant;
    private string $description;
    private string $type;
    private string $dateTransaction;
    private int $idUtilisateur;
    private int $idCategorie;
    private ?string $nomCategorie = null;

    public function getIdTransaction(): ?int
    {
        return $this->idTransaction;
    }

    public function setIdTransaction(?int $idTransaction): void
    {
        $this->idTransaction = $idTransaction;
    }

    public function getMontant(): float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): void
    {
        $this->montant = $montant;
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function setType(string $type): void
    {
        $this->type = $type;
    }

    public function getDateTransaction(): string
    {
        return $this->dateTransaction;
    }

    public function setDateTransaction(string $dateTransaction): void
    {
        $this->dateTransaction = $dateTransaction;
    }

    public function getIdUtilisateur(): int
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(int $idUtilisateur): void
    {
        $this->idUtilisateur = $idUtilisateur;
    }

    public function getIdCategorie(): int
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(int $idCategorie): void
    {
        $this->idCategorie = $idCategorie;
    }

    public function getNomCategorie(): ?string
    {
        return $this->nomCategorie;
    }

    public function setNomCategorie(?string $nomCategorie): void
    {
        $this->nomCategorie = $nomCategorie;
    }
}

File: JAVA/final-BT-java/src/models/Transaction.java
package models;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Transaction {
    private int idTransaction;
    private BigDecimal montant;
    private LocalDate dateTransaction;
    private String description;
    private int idUtilisateur;
    private int idCategorie;
    private LocalDateTime dateCreation;
    private TypeCategorie type;
    private String nomCategorie;

    public Transaction() {
    }

    public Transaction(int idTransaction, BigDecimal montant, LocalDate dateTransaction,
            String description, int idUtilisateur, int idCategorie, LocalDateTime dateCreation, TypeCategorie type,
            String nomCategorie) {
        this.idTransaction = idTransaction;
        this.montant = montant;
        this.dateTransaction = dateTransaction;
        this.description = description;
        this.idUtilisateur = idUtilisateur;
        this.idCategorie = idCategorie;
        this.dateCreation = dateCreation;
        this.type = type;
        this.nomCategorie = nomCategorie;
    }

    @Override
    public String toString() {
        return idTransaction + " | "+ dateTransaction+ " | "+ nomCategorie+ " | "+ montant + " FCFA";
    }

    public int getIdTransaction() {
        return idTransaction;
    }

    public void setIdTransaction(int idTransaction) {
        this.idTransaction = idTransaction;
    }

    public BigDecimal getMontant() {
        return montant;
    }

    public void setMontant(BigDecimal montant) {
        this.montant = montant;
    }

    public LocalDate getDateTransaction() {
        return dateTransaction;
    }

    public void setDateTransaction(LocalDate dateTransaction) {
        this.dateTransaction = dateTransaction;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getIdUtilisateur() {
        return idUtilisateur;
    }

    public void setIdUtilisateur(int idUtilisateur) {
        this.idUtilisateur = idUtilisateur;
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    public LocalDateTime getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDateTime dateCreation) {
        this.dateCreation = dateCreation;
    }

    public TypeCategorie getType() {
        return type;
    }

    public void setType(TypeCategorie type) {
        this.type = type;
    }

    public String getNomCategorie() {
        return nomCategorie;
    }

    public void setNomCategorie(String nomCategorie) {
        this.nomCategorie = nomCategorie;
    }
}